### PR TITLE
Colorado Income Qualified Senior Housing Income Tax Credit

### LINKED ISSUE #2773: Colorado Income Qualifed Senior Housing Income Tax Credit
Reference: https://tax.colorado.gov/income-qualified-senior-housing-income-tax-credit

### PR DESCRIPTION
**Pull request by Weijia-Wang341**
Fixes #2773

### Summary
📄🏠👵🏼

This pull request adds new parameters and variables for the Colorado Income Qualified Senior Housing Income Tax Credit and the New Mexico standard deduction. These changes allow the policy engine to model these state-specific tax provisions and their effects on filers. The changes involve creating YAML files for the parameter values and brackets, and Python files for the variable formulas and metadata.

> _`YAML` files define_
> _Colorado and New Mexico_
> _credits, deductions_

### Walkthrough
*  Add Colorado Income Qualified Senior Housing Income Tax Credit for different filing statuses ([link](https://github.com/PolicyEngine/policyengine-us/pull/2824/files?diff=unified&w=0#diff-4ac36c3f043a9c5195aa2fe407e6caf143ce0d0377ea566a09575b2c9dc6428bL1-R20), [link](https://github.com/PolicyEngine/policyengine-us/pull/2824/files?diff=unified&w=0#diff-cd0968a343803974d37f728be015e9ac31c2923b69b123b605182dfb8af1e85dL1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2824/files?diff=unified&w=0#diff-bdfb25364590f0b37ed7521b98405bd45815f58b082b14e8af8a9429700a8355L1-R20))
* Add New Mexico standard deduction and extra standard deduction for elderly and blind filers ([link](https://github.com/PolicyEngine/policyengine-us/pull/2824/files?diff=unified&w=0#diff-5e2ba819d6c6bdb2c11a063a14ed7093511c708cf06e671d7af5daaa6f232e0eR1-R27), [link](https://github.com/PolicyEngine/policyengine-us/pull/2824/files?diff=unified&w=0#diff-232981c27a5f3924253100bb3ccfc6d6b2297f6a14a92334a3808110b9e09e17R1-R23))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml:1]: need to make a head_of_household.yaml, joint.yaml, widow.yaml

File: policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+description: CO Income Qualified Senior Housing Income Tax Credit
+metadata:
+  unit: currency-USD
+  type: Single, Married filing joint, Head of Household
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml:4]:
```suggestion
  type: single_amount
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  period: year
+  reference:
+    - title: Income Qualified Senior Housing Income Tax Credit
+      href: https://tax.colorado.gov/income-qualified-senior-housing-income-tax-credit
+    - title: 2022 Colorado Individual Income Tax Filing Guide
+      href: https://tax.colorado.gov/sites/tax/files/documents/DR_104_Book_2022.pdf#page=17
+   
+brackets:
+  - threshold:
+      2022-01-01: 25_499
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml:15]: below make it 0

Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/single.yaml:15]:
```suggestion
      2022-01-01: 25_500
```

File: policyengine_us/variables/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/co_Income_Qualified_Senior_Housing.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/co_Income_Qualified_Senior_Housing.py:1]: cant use capital letters

File: policyengine_us/variables/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/co_Income_Qualified_Senior_Housing.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from policyengine_us.model_api import *
+
+
+class co_income_qualified_senior_housing(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Colorado senior citizen credit"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/co/tax/income/credits/Income_Qualified_Senior_Housing/co_Income_Qualified_Senior_Housing.py:7]:
```suggestion
    label = "ColoradoIncome Qualified Senior Housing Income Tax Credit"
```

**Review by MaxGhenis — COMMENTED** [review]
why new mexico changes?

File: policyengine_us/parameters/gov/states/nm/tax/deductions/standard/base_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/deductions/standard/base_amount.yaml:1]: run a git pull upstream master and if these nm changes still show, delete duplicate files

File: policyengine_us/parameters/gov/states/co/tax/income/credits/income_qualified_senior_housing/widow.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  period: year
+  reference:
+    - title: Income Qualified Senior Housing Income Tax Credit
+      href: https://tax.colorado.gov/income-qualified-senior-housing-income-tax-credit
+    - title: 2022 Colorado Individual Income Tax Filing Guide
+      href: https://tax.colorado.gov/sites/tax/files/documents/DR_104_Book_2022.pdf#page=17
+   
+brackets:
+  - threshold:
+      2022-01-01: 25_500
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/income_qualified_senior_housing/widow.yaml:15]: need 2021 values and 2021 tax form reference

**Pull request closed without merging**